Ombi.Api/TheMovieDbApi.cs

```diff
@@ -100,5 +105,48 @@ public TmdbMovieDetails GetMovieInformationWithVideos(int tmdbId)
             var movies = await Client.GetMovie(imdbId);
             return movies ?? new Movie();
         }
+
+        public async Task<List<Movie>> SearchActor(string searchTerm)
+        {
+            return await SearchActor(searchTerm, null);
+        }
+
+        public async Task<List<Movie>> SearchActor(string searchTerm, Func<int, string, string, Task<bool>> alreadyAvailable)
+        {
+            SearchContainer<SearchPerson> result = await Client.SearchPerson(searchTerm);
+            
+            var people = result?.Results ?? new List<SearchPerson>();
+            var person = (people.Count != 0 ? people[0] : null);
+            var movies = new List<Movie>();
+            var counter = 0;
+            try
+            {
+                if (person != null)
+                {
+                    var credits = await Client.GetPersonMovieCredits(person.Id);
+                    //only get the first 10 movies and delay a bit between each request so we don't overload the API
+                    foreach (var credit in credits.Cast)
+                    {   if (counter == 10)
+                            break;
+                        if (alreadyAvailable == null || !(await alreadyAvailable(credit.Id, credit.Title, credit.ReleaseDate.Value.Year.ToString())))
+                        {
+                            movies.Add(await GetMovie(credit.Id));
+                            counter++;
+                        }
+                        else
+                        {
+                            //if we didn't add the movie, delay longer since we know we'll be making additional API calls
+                            await Task.Delay(75);
```

Comment:
Yeah.  It's a bit arbitrary.  Basically, if we are making 10 calls, we sleep 50ms.  If we make more than 10, we sleep 125ms for every extra call after that.

This definitely slows the search a bit and I'd love to lower the delays.  Maybe I'll rip that extra 75ms out, there's not really any rhyme or reason behind it -- just didn't wanna beat the hell out of the API and hit the rate limit.

1 - What is the API rate limit?
2 - Is the same API key being used across everyone's installation of Ombi?